armi/reactor/tests/test_assemblies.py

```diff
@@ -1137,6 +1137,44 @@ def test_carestianCoordinates(self):
         )
         self.assertEqual(a.coords(), (2.0, 2.0))
 
+    def test_pinPlenumVolume(self):
+        """
+        Test the volume of a pin in the assembly's plenum.
+        """
+        assembly = assemblies.Assembly("dummy")
+        self.assertEqual(0.0, assembly.getPinPlenumVolumeInCubicMeters())
+
+        pinPlenumVolume = 7.853981633974482e-06
+        testAssemblies = buildTestAssemblies()
+        plenumBlock = makeTestAssembly(
+            1, 2, grids.CartesianGrid.fromRectangle(2.0, 2.0))
+        plenumBlock.setType("plenum", Flags.PLENUM)
+        self.Assembly.setBlockMesh([10.0, 20.0, 30.0], conserveMassFlag="auto")
+        plenumBlock.append(testAssemblies[0])
```

Comment:
This is pretty interesting; you are appending an assembly to another assembly. I think lots of stuff in the ARMI composite model will sort of "work," but might do some surprising things. Assemblies expect to be collections of Blocks, so another way to go might be to just construct one in a similar fashion as is done in `buildTestAssemblies()`, except with some plenum components, then append that to `testAssemblies[0]` and use it for the test. An example of what a typical plenum block might look like can be found here: https://github.com/terrapower/armi/blob/5700728b129ef87ecbf01f4fef2fa82f582a92ac/armi/tests/tutorials/anl-afci-177-blueprints.yaml#L163

Reply:
As in to add an `isinstance(obj, Block)` check on `Assembly.add()`? Might be a pretty good idea. I'm all for firming up/formalizing all of the mysterious assumptions we make. If you were to add such a check, and the tests pass, I'd be happy to adopt it. I would be curious to know if some things stub their toe on such a check...